core/src/main/scala/org/broadinstitute/dsde/rawls/resourcebuffer/ResourceBufferService.scala

```diff
@@ -0,0 +1,37 @@
+package org.broadinstitute.dsde.rawls.resourcebuffer
+
+import java.util.UUID
+
+import org.broadinstitute.dsde.rawls.dataaccess.resourcebuffer.ResourceBufferDAO
+import org.broadinstitute.dsde.rawls.model.ProjectPoolType.ProjectPoolType
+import org.broadinstitute.dsde.rawls.model.{GoogleProjectId, PoolId, ProjectPoolId, ProjectPoolType, UserInfo}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ResourceBufferService {
+  def constructor(resourceBufferDAO: ResourceBufferDAO)(userInfo: UserInfo)(implicit executionContext: ExecutionContext): ResourceBufferService = {
+    new ResourceBufferService(resourceBufferDAO, userInfo)
+  }
+}
+class ResourceBufferService(resourceBufferDAO: ResourceBufferDAO, protected val userInfo: UserInfo) {
+
+  def GetGoogleProjectFromRBS(projectPoolType: ProjectPoolType): Future[GoogleProjectId] = getGoogleProjectFromRBS(projectPoolType)
+
+  def getGoogleProjectFromRBS(projectPoolType: ProjectPoolType = ProjectPoolType.Regular): Future[GoogleProjectId] = {
+
+    val projectPoolId: ProjectPoolId = resourceBufferDAO.getProjectPoolId(projectPoolType)
+
+    val handoutRequestId = generateHandoutRequestId(userInfo, projectPoolId)
+
+    resourceBufferDAO.handoutGoogleProject(PoolId(projectPoolId.value), handoutRequestId)
+  }
+
+  //  handoutRequestId:
+  //        The unique identifier presented by the client for a resource request.
+  //        Using the same handoutRequestId in the same pool would get the same resource back.
+  private def generateHandoutRequestId(userInfo: UserInfo, projectPoolId: ProjectPoolId): String = {
```

Comment:
It might be worth making a case class for this too so that we aren't passing Strings around